db/upgrade.php

```diff
@@ -628,5 +628,32 @@ function xmldb_attendance_upgrade($oldversion=0) {
 
     }
 
+    if ($oldversion < 2019080802) {
+
+        // Changing nullability of field sessiondetailspos on table attendance to not null.
+        $table = new xmldb_table('attendance');
+        $field = new xmldb_field('sessiondetailspos', XMLDB_TYPE_CHAR, '5', null, XMLDB_NOTNULL, null, 'left', 'subnet');
+
+        // Launch change of nullability for field sessiondetailspos.
+        $dbman->change_field_notnull($table, $field);
+
+        // Changing the default of field preventsharedip on table attendance_sessions to 1.
+        $table = new xmldb_table('attendance_sessions');
+        $field = new xmldb_field('preventsharedip', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, '1', 'absenteereport');
+
+        // Launch change of default for field preventsharedip.
+        $dbman->change_field_default($table, $field);
+
+        // Changing the default of field maxwarn on table attendance_warning to drop it.
+        $table = new xmldb_table('attendance_warning');
+        $field = new xmldb_field('maxwarn', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null, 'warnafter');
```

Comment:
This looks weird? - the default was incorrectly set at some point, but it was fixed in an upgrade step and the default is different here from what is defined in install.xml.

I wonder if the site you're working on has been upgraded incorrectly at some point.